### PR TITLE
v1.13 Backports 2023-10-11

### PR DESCRIPTION
**Pull request by joamaki**
* [x] #28355 (@giorio94)
 * [x] #28364 (@aanm)
 * [x] #28485 (@pchaigno)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 28355 28364 28485
```

### CODE REVIEW

**Review by aanm — APPROVED** [review]

File: .github/workflows/tests-clustermesh-upgrade.yaml [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+          fi
+
+          CILIUM_DOWNGRADE_VERSION=$(contrib/scripts/print-downgrade-version.sh)
+
+          CILIUM_IMAGE_SETTINGS=" \
+            --chart-directory=./cilium-newest/install/kubernetes/cilium \
+            --set=image.override=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/cilium-ci:${SHA} \
+            --set=operator.image.override=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/operator-generic-ci:${SHA} \
+            --set=clustermesh.apiserver.image.override=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/clustermesh-apiserver-ci:${SHA} \
+            --set=clustermesh.apiserver.kvstoremesh.image.override=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/kvstoremesh-ci:${SHA} \
```
Comment (giorio94) [inline .github/workflows/tests-clustermesh-upgrade.yaml]: Could you please drop all references to kvstoremesh from this commit, as kvstoremesh didn't exist in v1.13 and earlier?

**Review by pchaigno — APPROVED** [review]
My PR looks good. Thanks!

**Review by giorio94 — APPROVED** [review]
My PR looks good, thanks! I've started the CI to check if the new workflow completes correctly. 

Edit: all green :tada:

**Comment by joamaki** [comment]
/test-backport-1.13

**Review by squeed — APPROVED** [review]

**Comment by joamaki** [comment]
/test-1.19-4.19

**Pull request merged**